### PR TITLE
Gb/batch dict

### PR DESCRIPTION
**Pull request by grantbuster**

### CODE REVIEW

**Comment by grantbuster** [comment]
implements #231

File: conda.recipe/meta.yaml [hunk 1]
```diff
@@ -16,7 +16,7 @@ requirements:
     - pip
     - setuptools
     - click
-    - h5py
+    - h5py>=3.1.0
```
Comment (MRossol) [inline conda.recipe/meta.yaml:19]: I tried this in rex and it failed to build, turns out Conda is still at 2.1

File: reV/batch/batch.py [hunk 2]
```diff
@@ -295,15 +295,41 @@ def sub_dirs(self):
...
+            Value to be written to a batch job json file
+
+        Returns
+        -------
+        arg : str | int | float | dict
+            Cleaned value ready to be written to a batch job json file.
+        """
+
+        if isinstance(arg, str):
+            if '{' in arg and '}' in arg:
```
Comment (MRossol) [inline reV/batch/batch.py:317]: What does the {/} have to do with replacing ' for "?

**Review by MRossol — CHANGES_REQUESTED** [review]
One question, and we need to remove h5py version from the conda.recipe/meta.yaml

Also, I have a bug_fix PR relation to a rex release. We should coordinate merging these two PRs...

File: conda.recipe/meta.yaml (hunk 1 above)
Comment (grantbuster) [reply conda.recipe/meta.yaml]: ah i see. Fixed

File: reV/batch/batch.py (hunk 2 above)
Comment (grantbuster) [reply reV/batch/batch.py:317]: the braces look for a dictionary that has been stringified. The ' vs " replacement has to do with how json wants strings with the double quotes.

**Comment by grantbuster** [comment]
@MRossol removed the h5py version from conda. Let me know how you want to merge the two PR's. I incremented the rev version on this branch. Both should be easy to rebase.

**Comment by MRossol** [comment]
Lets merge this one and then I can rebase mine while the tests run

**Pull request merged**